classes/note/NoteDAO.inc.php

```diff
@@ -81,45 +86,55 @@ public function getByUserId($userId, $rangeInfo = null)
      */
-    public function getByAssoc($assocType, $assocId, $userId = null, $orderBy = self::NOTE_ORDER_DATE_CREATED, $sortDirection = self::SORT_DIRECTION_DESC, $isAdmin = false)
+    public function getByAssoc(int $assocType, int $assocId, ?int $userId = null, bool $isAdmin = false, 
+        bool $orderData = true, int $orderBy = self::NOTE_ORDER_DATE_CREATED, int $sortDirection = self::SORT_DIRECTION_DESC) : LazyCollection
     {
-        $params = [(int) $assocId, (int) $assocType];
+        $query = DB::table('notes')
+            ->where('assoc_id', '=', $assocId)
+            ->where('assoc_type', '=', $assocType);
+
         if ($userId) {
-            $params[] = (int) $userId;
+            $query->where('user_id', '=', $userId);
         }
 
-        // Sanitize sort ordering
-        switch ($orderBy) {
-            case self::NOTE_ORDER_ID:
-                $orderSanitized = 'note_id';
-                break;
-            case self::NOTE_ORDER_DATE_CREATED:
-            default:
-                $orderSanitized = 'date_created';
+        if (!$isAdmin) {
+            $query->where(function ($query) {
+                $query->whereNotNull('title')
+                    ->orWhereNotNull('contents');
+            });
         }
-        switch ($sortDirection) {
-            case self::SORT_DIRECTION_ASC:
-                $directionSanitized = 'ASC';
-                break;
-            case self::SORT_DIRECTION_DESC:
-            default:
-                $directionSanitized = 'DESC';
+
+        if ($orderData) {
+            // Sanitize sort ordering
+            switch ($orderBy) {
+                case self::NOTE_ORDER_ID:
+                    $orderSanitized = 'note_id';
+                    break;
+                case self::NOTE_ORDER_DATE_CREATED:
+                default:
+                    $orderSanitized = 'date_created';
+            }
+            switch ($sortDirection) {
+                case self::SORT_DIRECTION_ASC:
+                    $directionSanitized = 'ASC';
+                    break;
+                case self::SORT_DIRECTION_DESC:
+                default:
+                    $directionSanitized = 'DESC';
+            }
+
+            $query->orderBy($orderSanitized, $directionSanitized);
         }
 
-        $result = $this->retrieve(
-            $sql = 'SELECT	*
-			FROM	notes
-			WHERE	assoc_id = ?
-				AND assoc_type = ?
-				' . ($userId ? ' AND user_id = ?' : '') .
-                ($isAdmin ? '' : '
-				AND (title IS NOT NULL OR contents IS NOT NULL)') . '
-			ORDER BY ' . $orderSanitized . ' ' . $directionSanitized,
-            $params
-        );
-        return new DAOResultFactory($result, $this, '_fromRow', [], $sql, $params); // Counted in QueriesGridCellProvider
+        $rows = $query->select(['*'])->get();
+        
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->_fromRow(get_object_vars($row));
```

Comment:
I'd recommend yielding associatively by the note ID.

Reply:
Changed to `yield $row->note_id => $this->_fromRow(get_object_vars($row));`